### PR TITLE
Feature/map annotations

### PR DESCRIPTION
**Pull request by DavidWrightOS**
# Map Annotations

- Created a custom map annotation and annotation detail view
- Synced the selection of an event cell with the selection of that events map annotation. This makes it easier for the user to see where each individual event is on the map since they only need to tap on the event cell to identify its map marker
- Updated the `category` attribute from a `String` to an `enum`, and set up a list of possible category types as well as annotation view configurations for each

Trello URL: **_https://trello.com/c/3L9wqhCS_**

### CODE REVIEW

File: Resfeber/Model/EventCategory.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        case .retail, .grocery: return #colorLiteral(red: 0.9999757409, green: 0.701993525, blue: 0.000208069614, alpha: 1)
+        case .bank: return #colorLiteral(red: 0.4352324307, green: 0.5098394752, blue: 0.7018731236, alpha: 1)
+        case .gasStation: return #colorLiteral(red: 0.1132080629, green: 0.627499342, blue: 0.9998752475, alpha: 1)
+        case .chargeStation: return #colorLiteral(red: 0.2034551501, green: 0.7804297805, blue: 0.34896487, alpha: 1)
+        case .home: return #colorLiteral(red: 0, green: 0.6823994517, blue: 0.9371369481, alpha: 1)
+        case .work, .education: return #colorLiteral(red: 0.6509636641, green: 0.4549258351, blue: 0.286247611, alpha: 1)
+        case .medical, .pharmacy, .emergency, .hospital: return #colorLiteral(red: 1, green: 0.3647276163, blue: 0.3529253006, alpha: 1)
+        }
+    }
+}
```
Comment (johnnydhicks) [inline Resfeber/Model/EventCategory.swift]: That's a lot of decimal points haha. I think you'd probably want to create a nice way to instantiate a new color by passing in the values on a scale of zero to 255. So maybe inside of your RFColor class you could do something like: 
```
extension UIColor {
    static func rgba(r: Int, g: Int, b: Int, a: Double) -> UIColor {
        return UIColor(red: CGFloat(r)/255.0, green: CGFloat(g)/255.0, blue: CGFloat(b)/255.0,   alpha: CGFloat(a))
    }
}
```

That way you could do something like:
```
 case .medical, .pharmacy, .emergency, .hospital: return UIColor.rgba(r: 255, g: 93, b: 90, a: 1)
```

File: Resfeber/View Controllers/TripDetailViewController.swift [hunk 2]
```diff
@@ -260,20 +259,31 @@ extension TripDetailViewController: UICollectionViewDelegate {
...
         guard let cell = collectionView.cellForItem(at: indexPath) as? EventCell,
               let event = cell.event else { return }
-
-        let detailVC = EventDetailViewController(event)
-        show(detailVC, sender: self)
+        
+        mapView.selectAnnotation(event, animated: true)
+        
+//        let detailVC = EventDetailViewController(event)
+//        show(detailVC, sender: self)
```
Comment (johnnydhicks) [inline Resfeber/View Controllers/TripDetailViewController.swift]: Get rid of these comments

File: Resfeber/View Controllers/TripDetailViewController.swift [hunk 3]
```diff
@@ -302,7 +312,52 @@ extension TripDetailViewController: UITableViewDelegate {
...
 }
+
+extension TripDetailViewController: MKMapViewDelegate {
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        guard let event = annotation as? Event else { return nil }
+        
+        guard let annotationView = mapView.dequeueReusableAnnotationView(withIdentifier: Event.annotationReuseIdentifier, for: event) as? MKMarkerAnnotationView else {
+            NSLog("Missing the registered map annotation view")
+            return nil
+        }
```
Comment (johnnydhicks) [inline Resfeber/View Controllers/TripDetailViewController.swift]:
```
        guard let event = annotation as? Event,
         let annotationView = mapView.dequeueReusableAnnotationView(withIdentifier: Event.annotationReuseIdentifier, for: event) as? MKMarkerAnnotationView else {
            NSLog("Missing the registered map annotation view")
            return nil
        }
```

**Review by johnnydhicks — CHANGES_REQUESTED** [review]

File: Resfeber/Model/EventCategory.swift (hunk 1 above)
Comment (DavidWrightOS) [reply Resfeber/Model/EventCategory.swift]: Implemented the above suggestion in a followup commit

File: Resfeber/View Controllers/TripDetailViewController.swift (hunk 3 above)
Comment (DavidWrightOS) [reply Resfeber/View Controllers/TripDetailViewController.swift]: Implemented the above suggestion in a followup commit

File: Resfeber/View Controllers/TripDetailViewController.swift (hunk 2 above)
Comment (DavidWrightOS) [reply Resfeber/View Controllers/TripDetailViewController.swift]: Deleted lines in a followup commit

**Pull request merged**